app/attributes/attributecontroller.cpp

```diff
@@ -290,15 +301,14 @@ void AttributeController::updateOnLayerChange()
         root->setColumnCount( 1 );
       }
 
+      mHasTabs = evaluateHasTabs( root );
       for ( QgsAttributeEditorElement *element : root->children() )
       {
         if ( element->type() == QgsAttributeEditorElement::AeTypeContainer )
         {
           QgsAttributeEditorContainer *container = static_cast<QgsAttributeEditorContainer *>( element );
-          if ( !container->isGroupBox() )
+          if ( !container->isGroupBox() && mHasTabs )
```

Comment:
this can be checked directly in the before the loop over elements again.

```
if (allowTabs) {
  same as before
} else 
  createTab( root );
```